compiler/rustc_infer/src/infer/error_reporting/mod.rs

```diff
@@ -2879,6 +2880,23 @@ impl<'tcx> ObligationCauseExt<'tcx> for ObligationCause<'tcx> {
             _ => "types are compatible",
         }
     }
+
+    fn as_requirement_localised(&self) -> &'static str {
+        use crate::traits::ObligationCauseCode::*;
+        match self.code() {
+            CompareImplItemObligation { kind: ty::AssocKind::Fn, .. } => "method_compat",
```

Comment:
Would it be possible to transform these into the strings-to-be-matched-against in an `IntoDiagnosticArg` impl or something like that? Similarly for other types you do this with.

Reply:
Noted, will do.